src/client/app.coffee

```diff
@@ -269,6 +319,28 @@ onMouseMove = (e) ->
     mouseVector.y = 1 - 2 * ( e.clientY / window.innerHeight )
 
 
+canCast = (skill) ->
+    if gameView.actionPointsRemaining >= SkillBarrier.cost
+        return true
+    else
+        return false
+
+onKeyPress = (e) ->
+    console.log("KEY: ", e.keyCode)
+
+    if e.keyCode == 98 #B
+        if canCast(SkillBarrier)
+            raycaster.setFromCamera( mouseVector, camera )
+
+            intersects = raycaster.intersectObjects(hexagons.children)
+            if intersects.length > 0
+                hexUuid = intersects[0].object.uuid
+                clickedHex = uuidToHex.get hexUuid
+                barrierSkill = new SkillBarrier()
+                castSkill(clickedHex, barrierSkill)
+                gameView.actionPointsRemaining -= SkillBarrier.cost
```

Comment:
this should be in game view so you can actually check if you are going over the cost